src/core/components/ErrorPage/index.js

```diff
@@ -36,6 +36,7 @@ export class ErrorPageBase extends React.Component<InternalProps> {
 
   render() {
     const { children, errorPage, getErrorComponent } = this.props;
+    invariant(getErrorComponent, 'getErrorComponent() is undefined');
```

Comment:
So is the pattern "if a prop has a default value and the prop is not for testing purpose only, we mark it as optional and we add an `invariant` call"?

Reply:
Yes.

Reply:
I'm confused. If it has a default, why do we need the `invariant` at all? In case someone explicitly passes `undefined` into `getErrorComponent`? Or is this to deal with a Flow issue?

Reply:
Flow normally handles `defaultProps` just fine. It sees a required prop but then knows it can be left undefined because of the default value.

In https://github.com/mozilla/addons-frontend/issues/3563 I wanted to leverage Flow to help us validate component properties for those components wrapped in HOCs. Before any of the patches to address it, we were completely flying blind. You could instantiate any HOC wrapped component with invalid props or missing props and Flow would not complain.

The solution was not straight forward. It consists of manually declaring a type hint for the final exported component. However, it only requires one extra line of code and it will validate usage of props.

A side effect of manually declaring a type hint is that Flow loses the ability to infer `defaultProps`. This means we will now need to declare these props as optional even though internally they can be handled like required props. Using invariant should make it pretty easy to deal with this change.